### ISSUE TITLE
Feature: masked slicing

### ISSUE THREAD

**Issue by arsserpentarium** [ilanschnell/bitarray#190]
there is other quite similar feature suggestion,which is related to list indexing:
https://github.com/ilanschnell/bitarray/issues/156
However, I need a bit different thing. List indexing for bit retrieval:
`
>>> a = bitarray('10010011')
>>> lst = [1,0,1,0,1,1,1,1] #masking
>>>a[lst]
bitarray('100011')
>>>a[bitarray('10111000')]
bitarray('1101')
>>> lst = [0,1,3,7] #indexing
>>>a[lst]
bitarray('1011')
`

I'd be glad to see that functions in any implementation.

**Comment by ilanschnell** [comment]
Thanks for using bitarray!  If I understand your example correctly, masking is simply the (already existing) `&` operator:
```
>>> a = bitarray('10010011')
>>> lst = [1,0,1,0,1,1,1,1] #masking
>>> a & bitarray(lst)  # you wrote a[lst] here
bitarray('10000011')
```
I'm having the same concerns as with #156 for adding the *indexing* with a list feature.  So having `bitarray.util.getitems()` as well.  If I get some time, I would implement both of these functions at once.

**Comment by arsserpentarium** [comment]
nope. And operator can't compose bitarray from bits of sequence a, which are on places of 1 in b. Numpy can do such thing, but it's quite inconvenient to use numpy for binary processing.

It'll be fine if that feature would be implemented as function.

thank you for your library.

**Comment by ilanschnell** [comment]
#204 has now been merged which allows indexing by lists

**Comment by ilanschnell** [comment]
#205 has been merged which can now allows indexing by bitarrays as masks, see:
https://github.com/ilanschnell/bitarray/blob/master/doc/indexing.rst#masked-indexing

**Comment by addandaum** [comment]
Oh thank you very much.

**Comment by ilanschnell** [comment]
The new features (sequence and masked indexing) are now available in bitarray 2.8.0
